client/src/components/formHelpers/form-fields.tsx

```diff
@@ -115,8 +115,9 @@ function FormFields({ formFields, options }: FormFieldsProps): JSX.Element {
                     placeholder={placeholder}
                     required={required.includes(name)}
                     rows={4}
-                    type='url'
+                    type={types[name] || 'text'}
```

Comment:
If we start allowing dynamic types, the `types` definition should be made stricter.

From:

```ts
types?: { [key: string]: string };
```

To:

```ts
types?: { [key: string]: React.HTMLInputTypeAttribute };
```